File: backend/src/main/java/com/nexora/service/ProjectService.java
package com.nexora.service;

import com.nexora.dto.CreateProjectRequest;
import com.nexora.dto.ProjectDTO;
import com.nexora.entity.Project;
import com.nexora.entity.User;
import com.nexora.repository.ProjectRepository;
import com.nexora.repository.UserRepository;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@Slf4j
public class ProjectService {

    @Autowired
    private ProjectRepository projectRepository;

    @Autowired
    private UserRepository userRepository;

    @Transactional
    public ProjectDTO createProject(CreateProjectRequest request, Long userId) {
        User user = userRepository.findById(userId)
                .orElseThrow(() -> new RuntimeException("User not found"));

        if (projectRepository.existsByNameAndUserId(request.getName(), userId)) {
            throw new RuntimeException("Project with this name already exists");
        }

        Project project = Project.builder()
                .user(user)
                .name(request.getName())
                .description(request.getDescription())
                .repositoryUrl(request.getRepositoryUrl())
                .isActive(true)
                .build();

        projectRepository.save(project);
        log.info("Project created: {} for user: {}", request.getName(), userId);
        return convertToDTO(project);
    }

    public List<ProjectDTO> getUserProjects(Long userId) {
        return projectRepository.findByUserIdAndIsActiveTrue(userId)
                .stream()
                .map(this::convertToDTO)
                .collect(Collectors.toList());
    }

    public ProjectDTO getProjectById(Long id, Long userId) {
        Project project = projectRepository.findByIdAndUserId(id, userId)
                .orElseThrow(() -> new RuntimeException("Project not found"));
        return convertToDTO(project);
    }

    @Transactional
    public ProjectDTO updateProject(Long id, CreateProjectRequest request, Long userId) {
        Project project = projectRepository.findByIdAndUserId(id, userId)
                .orElseThrow(() -> new RuntimeException("Project not found"));

        project.setName(request.getName());
        project.setDescription(request.getDescription());
        project.setRepositoryUrl(request.getRepositoryUrl());

        projectRepository.save(project);
        log.info("Project updated: {}", id);
        return convertToDTO(project);
    }

    @Transactional
    public void deleteProject(Long id, Long userId) {
        Project project = projectRepository.findByIdAndUserId(id, userId)
                .orElseThrow(() -> new RuntimeException("Project not found"));
        project.setIsActive(false);
        projectRepository.save(project);
        log.info("Project deleted: {}", id);
    }

    private ProjectDTO convertToDTO(Project project) {
        return ProjectDTO.builder()
                .id(project.getId())
                .name(project.getName())
                .description(project.getDescription())
                .repositoryUrl(project.getRepositoryUrl())
                .framework(project.getFramework())
                .language(project.getLanguage())
                .architecturePattern(project.getArchitecturePattern())
                .isActive(project.getIsActive())
                .build();
    }
}

File: frontend/src/services/projectService.ts
import api from './api';

export interface Project {
  id: number;
  name: string;
  description: string;
  repositoryUrl: string;
  framework?: string;
  language?: string;
  architecturePattern?: string;
  isActive: boolean;
}

export interface CreateProjectRequest {
  name: string;
  description: string;
  repositoryUrl: string;
}

const projectService = {
  getAllProjects: async (): Promise<Project[]> => {
    const response = await api.get('/projects');
    return response.data;
  },

  getProjectById: async (id: number): Promise<Project> => {
    const response = await api.get(`/projects/${id}`);
    return response.data;
  },

  createProject: async (request: CreateProjectRequest): Promise<Project> => {
    const response = await api.post('/projects', request);
    return response.data;
  },

  updateProject: async (id: number, request: CreateProjectRequest): Promise<Project> => {
    const response = await api.put(`/projects/${id}`, request);
    return response.data;
  },

  deleteProject: async (id: number): Promise<void> => {
    await api.delete(`/projects/${id}`);
  },
};

export default projectService;
